src/App.test.jsx

```diff
@@ -4,31 +4,75 @@ import { render } from '@testing-library/react';
 
 import { useDispatch, useSelector } from 'react-redux';
 
+import { MemoryRouter } from 'react-router-dom';
+
+import regions from '../fixtures/regions';
+import categories from '../fixtures/categories';
+import restaurants from '../fixtures/restaurants';
+import restaurant from '../fixtures/restaurant';
+
 import App from './App';
 
-test('App', () => {
+jest.mock('react-redux');
+
+describe('App', () => {
   const dispatch = jest.fn();
 
   useDispatch.mockImplementation(() => dispatch);
 
   useSelector.mockImplementation((selector) => selector({
-    regions: [
-      { id: 1, name: '서울' },
-    ],
-    categories: [
-      { id: 1, name: '한식' },
-    ],
-    restaurants: [
-      { id: 1, name: '마법사주방' },
-    ],
+    regions,
+    categories,
+    restaurants,
+    restaurant,
   }));
 
-  const { queryByText } = render((
-    <App />
-  ));
+  const renderApp = ({ path }) => render(
+    <MemoryRouter initialEntries={[path]}>
+      <App />
+    </MemoryRouter>,
+  );
+
+  context('with path /', () => {
+    it('renders home page', () => {
+      const { container } = renderApp({ path: '/' });
+
+      expect(container).toHaveTextContent('Home');
+      expect(container).toHaveTextContent('About');
+      expect(container).toHaveTextContent('Restaurants');
+    });
+  });
+
+  context('with path /about', () => {
+    it('renders about page', () => {
+      const { container } = renderApp({ path: '/about' });
+
+      expect(container).toHaveTextContent(/About 페이지/);
+    });
+  });
+
+  context('with path /restaurants', () => {
+    it('renders restaurants page', () => {
+      const { container } = renderApp({ path: '/restaurants' });
+
+      expect(container).toHaveTextContent(/서울/);
+      expect(container).toHaveTextContent(/한식/);
+    });
+  });
+
+  context('with path /restaurants/:id', () => {
+    it('renders restaurant page', () => {
+      const { container } = renderApp({ path: '/restaurants/1' });
+
+      expect(container).toHaveTextContent(/양천주가/);
+    });
+  });
 
-  expect(dispatch).toBeCalled();
+  context('with unknown path', () => {
+    it('renders Not Found page', () => {
+      const { container } = renderApp({ path: '/test' });
```

Comment:
```suggestion
      const { container } = renderApp({ path: '/xxxx' });
```

와 같이 직접적인 표현이면 좋을 것 같네요. test는 뭔가 있을 법한...ㅎㅎ